### PR TITLE
[vue][sims-#1311] Remove primevue per component and upgrade vuetify#1311 part 2

### PR DESCRIPTION
**Pull request by ann-aot**
- Last PR for this story, in this PR below primevue components are replaced,

1. Message
2. Checkbox
3. Calendar
4. Dropdown
5. ProgressSpinner
6. Chip
7. TabMenu
8. TabView
9. TabPanel
10. font-awesome-icon
11. Timeline
12. tooltip

- Refactored status components
- removed used styles
- added all Figma colors to vuetify
- working on `v-htm` SonarCloud error ->slots added
- `.v-tab__slider` override (implemented in https://github.com/bcgov/SIMS/pull/1335) is breaking other tabs -> fixed, added `stacked` props
- most of the changes are repeated logic.
-code smells due to TODO

**Screenshots**
image
image
image

### CODE REVIEW

File: sources/packages/web/package.json [hunk 1]
```diff
@@ -30,6 +26,7 @@
     "primevue": "^3.8.2",
     "roboto-fontface": "*",
     "s-vuex-class": "^0.4.1",
+    "sanitize-html": "^2.7.1",
```
Comment (andrewsignori-aot) [inline sources/packages/web/package.json:29]: As we talked lets have this package removed.

File: sources/packages/web/src/assets/css/base.scss [hunk 2]
```diff
@@ -203,3 +189,11 @@ html * {
 .v-text-field-search-width {
   width: 305px;
 }
+
+.success-chip-background {
+  background-color: $banner-background-success !important;
+}
+
+.warning-chip-background {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/assets/css/base.scss:197]: Are we planning to add the other ones as needed?

File: sources/packages/web/src/components/aest/OfferingChangeRequest.vue [hunk 3]
```diff
@@ -13,7 +13,7 @@ import {
   OfferingDTO,
 } from "@/types";
 import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
-import { BannerTypes } from "@/components/generic/Banner.models";
+import { BannerTypes } from "@/types/contracts/Banner";
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/OfferingChangeRequest.vue:16]: Thanks for moving the type. I believe that it can be included in the "@/types" index. In future reorganizations, it will cause less impact.

File: sources/packages/web/src/components/generic/Banner.vue [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
       <div class="label-bold-normal">{{ header }}</div>
     </template>
     <v-row>
-      <v-col sm="10" md="10" lg="10"
-        ><div class="label-value-normal">{{ summary }}</div></v-col
-      >
-      <v-col sm="2" md="2" lg="2"><slot name="actions"></slot></v-col>
+      <v-col cols="10">
+        <div class="label-value-normal" v-if="summary">{{ summary }}</div>
+        <div class="label-value-normal"><slot name="content"></slot></div>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/Banner.vue:9]: I believe that it can be simplified as below:
```html
<div class="label-value-normal"><slot name="content">{{ summary }}</slot></div>
```

File: sources/packages/web/src/components/generic/HeaderNavigator.vue [hunk 5]
```diff
@@ -4,7 +4,7 @@
       <slot name="title">
         <div v-if="routeLocation" class="header-title">
           <a @click="goBack()">
-            <font-awesome-icon :icon="['fas', 'arrow-left']" class="mr-2" />
+            <v-icon icon="fa:fa fa-arrow-left" size="25"></v-icon>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/HeaderNavigator.vue:7]: Is the `size="25"` still needed?

File: sources/packages/web/src/components/generic/StatusChip.vue [hunk 6]
```diff
@@ -24,42 +20,131 @@ export default {
       type: String,
       required: false,
     },
+    /**
+     * When property is true its is a normal status like chip
+     *  and when it is false it is a designation or restriction
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/StatusChip.vue:25]: There is an extra space in this sentence.

File: sources/packages/web/src/components/generic/StatusChip.vue [hunk 7]
```diff
@@ -24,42 +20,131 @@ export default {
       type: String,
       required: false,
     },
+    /**
+     * When property is true its is a normal status like chip
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/StatusChip.vue:24]: I am not following this part. The `StatusChip` is supposed to be generic. Why are there different behaviors for designation or restriction?

File: sources/packages/web/src/components/generic/StatusChipApplication.vue [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+</template>
+<script lang="ts">
+import { computed } from "vue";
+import StatusChip from "@/components/generic/StatusChip.vue";
+import { useApplication } from "@/composables";
+export default {
+  components: { StatusChip },
+  props: {
+    status: {
+      type: String,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/StatusChipApplication.vue:12]: Can we have these properties with a better type definition, like below or adding a validator or both?
```
props: {
    status: {
      type: String as PropType<ApplicationStatus>,
      required: true,
    },
  },
```

File: sources/packages/web/src/composables/useApplication.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+export function useApplication() {
+  const mapApplicationChipStatus = (
+    status: ApplicationStatus,
+  ): StatusChipTypes => {
+    switch (status) {
+      case ApplicationStatus.completed:
+        return StatusChipTypes.Success;
+      case ApplicationStatus.inProgress:
+        return StatusChipTypes.Warning;
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/composables/useApplication.ts:12]: The items returning the same status can be grouped.

File: sources/packages/web/src/plugins/vuetify.ts [hunk 10]
```diff
@@ -17,6 +17,22 @@ export default createVuetify({
         dark: false,
         colors: {
           primary: "#2965c5",
+          secondary: "#69768C",
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/plugins/vuetify.ts:20]: ❤️

File: sources/packages/web/src/types/contracts/RestrictionContract.ts [hunk 11]
```diff
@@ -86,3 +86,15 @@ export enum RestrictionNotificationType {
    */
   Error = "Error",
 }
+
+// Student restriction badge.
+export enum StudentRestrictionStatus {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/contracts/RestrictionContract.ts:91]: Please add the comments in a way that VS Code will recognize them.

File: sources/packages/web/src/types/contracts/RestrictionContract.ts [hunk 12]
```diff
@@ -86,3 +86,15 @@ export enum RestrictionNotificationType {
...
+  noRestriction = "No restrictions",
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/contracts/RestrictionContract.ts:92]: I am assuming that this was an existing enum that still needs to be changed to the most updated naming convention, right?

File: sources/packages/web/src/views/aest/institution/InstitutionDetails.vue [hunk 13]
```diff
@@ -30,81 +48,72 @@ export default {
     },
   },
   setup(props: any) {
-    const router = useRouter();
-
+    const tab = ref("institution-tab");
     const institutionBasicDetail = ref({} as InstitutionBasicAPIOutDTO);
     // TODO: replace all fa isons with fas as per figma with replace with vuetify3
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/InstitutionDetails.vue:53]: Can we remove this TODO now?

File: sources/packages/web/src/views/aest/institution/InstitutionNotes.vue [hunk 14]
```diff
@@ -1,55 +1,57 @@
...
-    </div>
-  </v-card>
+  <v-container>
+    <v-row class="my-2 pl-0">
+      <v-col cols="3" class="category-header-large">Notes</v-col>
+      <v-col class="text-center">
+        <div class="float-right">
+          <v-row>
+            <v-col>
+              <v-btn
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/InstitutionNotes.vue:9]: Is this not supposed to be a `v-btn-toggle` (https://next.vuetifyjs.com/en/components/button-groups/)?

File: sources/packages/web/src/views/aest/institution/InstitutionNotes.vue [hunk 15]
```diff
@@ -1,55 +1,57 @@
...
+                data-cy="noteTypeItem"
+                @click="filterNotes(item)"
+                >{{ item }}</v-btn
+              >
+            </v-col>
+          </v-row>
+        </div>
+      </v-col>
+    </v-row>
+    <full-page-container
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/InstitutionNotes.vue:30]: Not part of the PR but this page is not following the layout of the other ones, right? Probably a page from before the introduction of the components.

File: sources/packages/web/src/views/aest/institution/Restrictions.vue [hunk 16]
```diff
@@ -1,79 +1,76 @@
...
+            slotProps.data.isActive
+              ? "-"
+              : dateOnlyLongString(slotProps.data.updatedAt)
+          }}</template></Column
+        >
+        <Column field="isActive" header="Status">
+          <template #body="slotProps">
+            <status-chip-restriction
+              :status="
+                slotProps.data.isActive
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Restrictions.vue:53]: Is this transformation needed? Since the `status-chip-restriction` is specific to the restriction can it be directly receiving the `RestrictionStatus`?

File: sources/packages/web/src/views/aest/institution/Restrictions.vue [hunk 17]
```diff
@@ -1,79 +1,76 @@
...
-                >View</v-btn
-              >
-            </template></Column
-          >
-        </DataTable>
-      </content-group>
-    </div>
-  </v-card>
+        </Column>
+        <!-- TODO: the color attribute has to come from either global constant or styling needs to be added to added. -->
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Restrictions.vue:60]: Is this to-do still needed?

File: sources/packages/web/src/views/student/Login.vue [hunk 18]
```diff
@@ -12,17 +12,14 @@
       >We are using BCSC for authentication. Please click on Login/Register
       buttons below to start your sign in/sign up.</v-card-text
     >
-    <v-card-actions>
+    <!-- TODO: v-btn inside v-card-actions is not working properly. try in vuetify 3 stable version -->
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/Login.vue:15]: The next PR for student BCeID will refactor this and it will no longer be a problem.

File: sources/packages/web/src/assets/css/base.scss (hunk 2 above)
Comment (ann-aot) [reply sources/packages/web/src/assets/css/base.scss:197]: as per the design system, badge with background color only these 2, (success and warning, i have added them both)
image

File: sources/packages/web/src/components/generic/HeaderNavigator.vue (hunk 5 above)
Comment (ann-aot) [reply sources/packages/web/src/components/generic/HeaderNavigator.vue:7]: without the size, the icon is slightly big 
image
in Figma its slightly small,
image

File: sources/packages/web/src/components/generic/StatusChip.vue (hunk 7 above)
Comment (ann-aot) [reply sources/packages/web/src/components/generic/StatusChip.vue:24]: image

Comment (ann-aot) [reply sources/packages/web/src/components/generic/StatusChip.vue:24]: first 2 are for designation or restriction

**Review by andrewsignori-aot — COMMENTED** [review]
Great work, great refactors. Please take a look at the comments.

File: sources/packages/web/src/assets/css/base.scss (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/assets/css/base.scss:197]: Let's talk about it. I believe that maybe we can have these components organized in a different way.

File: sources/packages/web/src/types/contracts/RestrictionContract.ts (hunk 12 above)
Comment (ann-aot) [reply sources/packages/web/src/types/contracts/RestrictionContract.ts:92]: yes, it is. I will update

File: sources/packages/web/src/views/aest/institution/Restrictions.vue (hunk 16 above)
Comment (ann-aot) [reply sources/packages/web/src/views/aest/institution/Restrictions.vue:53]: here `isActive` is a boolean field. I believe the transformation logic is added to translate the boolean value to restriction status

File: sources/packages/web/src/views/aest/institution/Restrictions.vue (hunk 17 above)
Comment (ann-aot) [reply sources/packages/web/src/views/aest/institution/Restrictions.vue:60]: removed

File: sources/packages/web/src/components/generic/StatusChip.vue [hunk 19]
```diff
@@ -1,18 +1,14 @@
 <template>
-  <v-chip :text-color="textColor" :class="chipClass" variant="outlined">
-    <v-icon start icon="fa:fa fa-circle" :color="iconColor" size="12" />
-    {{ label ?? status }}
+  <v-chip :color="chipColor" variant="outlined" :class="chipBackground"
+    ><v-icon :color="iconColor" :icon="icon" size="18" />
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/components/generic/StatusChip.vue:3]: Is the size="18" the same as what we have in figma ?

File: sources/packages/web/src/components/students/StudentDocumentList.vue [hunk 20]
```diff
@@ -12,8 +12,8 @@
           @click="fileUtils.downloadStudentDocument(ministryDocument)"
         >
           <span class="mr-4">
-            <font-awesome-icon :icon="['far', 'file-alt']"
-          /></span>
+            <v-icon icon="fa:far fa-file-alt" size="20"></v-icon
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/components/students/StudentDocumentList.vue:15]: Do we have a consistent icon size or are there different sizes in figma. In the other comment I made, the size is 18.

File: sources/packages/web/src/components/students/StudentDocumentList.vue [hunk 21]
```diff
@@ -36,8 +36,8 @@
           @click="fileUtils.downloadStudentDocument(studentDocument)"
         >
           <span class="mr-4">
-            <font-awesome-icon :icon="['far', 'file-alt']"
-          /></span>
+            <v-icon icon="fa:far fa-file-alt" size="20"></v-icon
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/components/students/StudentDocumentList.vue:39]: Please see previous comments on the size.

File: sources/packages/web/src/components/aest/StudentApplications.vue [hunk 22]
```diff
@@ -76,33 +81,41 @@
...
-                variant="plain"
-              >
-                <font-awesome-icon
-                  :icon="['fas', 'trash']"
-                  v-tooltip="'Click To Cancel this Application'"
-                  @click="$emit('openConfirmCancel', slotProps.data.id)"
-                />
-              </v-btn>
+              <v-tooltip>
+                <template v-slot:activator="{ props }">
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/aest/StudentApplications.vue:85]: Only for programmatic tool tip we need to wrap button inside <v-tooltip>. Otherwise e can define the the v-tooltip inside v-btn. Correct me if I am wrong.

File: sources/packages/web/src/components/aest/institution/modals/ApproveProgramModal.vue [hunk 23]
```diff
@@ -57,7 +56,6 @@ export default {
       approveProgram,
       showModal,
       dialogClosed,
-      COLOR_BLUE,
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/aest/institution/modals/ApproveProgramModal.vue:60]: 👍

File: sources/packages/web/src/components/aest/StudentApplications.vue (hunk 22 above)
Comment (ann-aot) [reply sources/packages/web/src/components/aest/StudentApplications.vue:85]: i didn't understand

File: sources/packages/web/src/components/students/StudentDocumentList.vue (hunk 20 above)
Comment (ann-aot) [reply sources/packages/web/src/components/students/StudentDocumentList.vue:15]: all the places where I have added size for the icon were big than the Figma, for eg.. 
1. below is the icon with default size
image
below is the figma,
image

File: sources/packages/web/src/components/generic/StatusChip.vue (hunk 19 above)
Comment (ann-aot) [reply sources/packages/web/src/components/generic/StatusChip.vue:3]: all the places where I have added size for the icon were big than the Figma, for eg.. 
1. below is the icon with default size
image
below is the figma,
image

File: sources/packages/web/src/components/students/StudentDocumentList.vue (hunk 20 above)
Comment (ann-aot) [reply sources/packages/web/src/components/students/StudentDocumentList.vue:15]: cmmented above

File: sources/packages/web/src/components/students/StudentDocumentList.vue (hunk 21 above)
Comment (ann-aot) [reply sources/packages/web/src/components/students/StudentDocumentList.vue:39]: pls see the above comment

File: sources/packages/web/src/components/aest/StudentApplications.vue (hunk 22 above)
Comment (ann-aot) [reply sources/packages/web/src/components/aest/StudentApplications.vue:85]: getting vue compilation error, 
image

Comment (ann-aot) [reply sources/packages/web/src/components/aest/StudentApplications.vue]: both methods are correct. updated to the one that use less code :)

**Review by henryoforeh-aot — APPROVED** [review]

**Review by guru-aot — APPROVED** [review]
LGTM, Nice work @ann-aot . Thats intense work 👍

File: sources/packages/web/src/views/student/StudentApplicationDetails.vue [hunk 24]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <student-page-container :full-width="true" layout-template="centered">
+  <student-page-container :fullWidth="true" layout-template="centered">
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/student/StudentApplicationDetails.vue:2]: Are we planning to use the camel case for attribute names?

File: sources/packages/web/src/components/generic/ChipLabel.vue [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+      type: String,
+      required: false,
+    },
+  },
+  setup(props: any) {
+    const chipColor = computed(() => {
+      switch (props.status) {
+        case StatusChipTypes.Success:
+          return "success";
+
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/ChipLabel.vue:27]: Please remove the extra blank lines.

File: sources/packages/web/src/components/generic/ChipStatus.vue [hunk 26]
```diff
@@ -26,40 +20,45 @@ export default {
     },
   },
   setup(props: any) {
-    const iconColor = computed(() => {
+    const chipColor = computed(() => {
       switch (props.status) {
         case StatusChipTypes.Success:
-          return COLOR_BANNER_SUCCESS;
+          return "success_bg";
+
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/ChipStatus.vue:27]: Please remove extra blank lines.

File: sources/packages/web/src/views/aest/institution/InstitutionNotes.vue [hunk 27]
```diff
@@ -1,55 +1,58 @@
...
-    </div>
-  </v-card>
+  <full-page-container
+    :layout-template="LayoutTemplates.Centered"
+    :fullWidth="true"
+  >
+    <v-row class="my-2 pl-0">
+      <v-col cols="3" class="category-header-large">Notes</v-col>
+      <v-col>
+        <v-btn-toggle
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/InstitutionNotes.vue:9]: Thanks for changing it ❤️

**Review by andrewsignori-aot — APPROVED** [review]
Great work, thanks for the changes and for searching for the components that fit better. Looks good 👍

File: sources/packages/web/src/components/institutions/active-application/ActiveApplicationSummaryData.vue [hunk 28]
```diff
@@ -1,85 +1,90 @@
 <template>
-  <body-header
-    title="Applications"
-    :recordsCount="applications.results?.length"
-    class="m-1"
+  <full-page-container
+    :layout-template="LayoutTemplates.CenteredCard"
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/institutions/active-application/ActiveApplicationSummaryData.vue:3]: CenteredCard is the default property value of layout-template. So not required to explicitly assign.

**Review by dheepak-aot — APPROVED** [review]
Great work @ann-aot 👍. 
2 minor comments which are not blockers.
